prinz/src/main/scala/pl/touk/nussknacker/prinz/util/collection/immutable/VectorMultimap.scala

```diff
@@ -0,0 +1,113 @@
+package pl.touk.nussknacker.prinz.util.collection.immutable
+
+import scala.collection.mutable
+
+class VectorMultimap[K, V](private val delegate: mutable.LinkedHashMap[K, Vector[V]]) {
+
+  def add(key: K, value: V): VectorMultimap[K, V] = get(key) match {
+    case None =>
+      val newVector = Vector(value)
+      val newMap = delegate.clone()
+      newMap.update(key, newVector)
+      VectorMultimap(newMap)
+    case Some(oldVector) =>
+      val newVector = oldVector :+ value
+      val newMap = delegate.clone()
+      newMap.update(key, newVector)
+      VectorMultimap(newMap)
+  }
+
+  def addAll(elements: Iterable[(K, V)]): VectorMultimap[K, V] = {
+    val created = delegate.clone()
+    elements.foreach { case (key, value) =>
+      created.get(key) match {
+        case None => created(key) = Vector(value)
+        case Some(vector) => created(key) = vector :+ value
+      }
+    }
+    VectorMultimap(created)
+  }
+
+  def exists(key: K, p: V => Boolean): Boolean = get(key) match {
+    case None => false
+    case Some(vector) => vector exists p
+  }
+
+  def mapValues[B](f: V => B): VectorMultimap[K, B] = {
+    val mapped = for (key <- keys.toList; value <- get(key).get)
```

Comment:
let's use multiline version for readability and to avoid ; 
for {
...
} yield